app/controllers/publishers_controller.rb

```diff
@@ -423,6 +423,43 @@ def balance
     end
   end
 
+  def banner_editor_data
+    # Handle edge case where user hasn't created channels yet.
+    no_channels_banner = current_publisher.site_banners.find_by(channel_id: "00000000-0000-0000-0000-000000000000")
+    if no_channels_banner && !current_publisher.channels.length.zero?
+      no_channels_banner.delete
+    end
+
+    channels = []
```

Comment:
433 to 440 can be refactored to `channels = current_publisher.channels.map { |channel| {id: channel.id, name: channel.publication_title, type: channel.details_type} }`

Reply:
Done ✔️